components/NotificationBar.js

```diff
@@ -117,6 +160,43 @@ class NotificationBar extends React.Component {
           <div className={`NotificationLine ${status}`}>
             <h1>{title}</h1>
             <p className="description">{description}</p>
+
+            {LoggedInUser.roles[host.slug] == 'ADMIN' && (
+              <AcceptReject>
+                <Fragment>
+                  <Mutation mutation={ApproveCollectiveMutation}>
+                    {(approveCollective, { loading }) => (
+                      <StyledButton
+                        m={1}
+                        loading={loading}
+                        onClick={() => approveCollective({ variables: { id: collective.id } })}
+                        data-cy={`${collective.slug}-approve-collective`}
+                        buttonStyle="primary"
+                        minWidth={125}
+                      >
+                        <FormattedMessage id="host.pending-applications.approve" defaultMessage="Approve" />
+                      </StyledButton>
+                    )}
+                  </Mutation>
+                  <StyledButton
+                    buttonStyle="danger"
+                    minWidth={125}
+                    m={1}
+                    onClick={() => this.setState({ showRejectionModal: true, collectiveId: collective.id })}
+                  >
+                    <FormattedMessage id="host.pending-applications.reject" defaultMessage="Reject" />
+                  </StyledButton>
+                </Fragment>
+                {this.state.showRejectionModal && (
+                  <AppRejectionReasonModal
+                    show={this.state.showRejectionModal}
+                    onClose={() => this.setState({ showRejectionModal: false })}
+                    collectiveId={this.state.collectiveId}
+                    hostCollectiveSlug={host.slug}
+                  />
+                )}
```

Comment:
Rather than duplicating this code from `components/host-dashboard/PendingApplications.js`, can we extract it to separate(s) component(s) to centralize the logic?

Reply:
Done